src/core/batchlan.cpp

```diff
@@ -1304,6 +1304,12 @@ HBLObjectRef       _ExecutionList::Execute     (_ExecutionList* parent, bool ign
 
     hy_env::EnvVariableSet(hy_env::path_to_current_bf, &cfp, true);
 
+    if (!hy_env::EnvVariableGet(hy_env::path_to_initial_bf, STRING)) {
+        hy_env::EnvVariableSet(hy_env::path_to_initial_bf, &cfp, true);
+        _String hy_relative_path = ((_FString *)hy_env::EnvVariableGet(hy_env::path_to_initial_bf, STRING))->get_str();
+        AppendLibraryPath(hy_relative_path);
```

Comment:
Need to make sure that we clean up library paths after the execution of individual files, otherwise the path stack will get very polluted. This should be done at the end of _ExecutionList::Execute

Reply:
Nice catch. I'll do that.